src/pipe.js

```diff
@@ -167,21 +167,29 @@
     }
     /*
     * Custom Performance entries that can be added from fragments
-    * Its need because Browsers currently does not allow expose any API to add
+    * It's needed because browsers currently do not expose an API to add
     * custom timing information to performance entries
     */
     function addPerfEntry(name, duration) {
         // Should not add to entries when Navigation timing is not supported.
         if (!'timing' in perf) {
             return;
         }
+        // duplicate entries are not handled to keep the API
+        // similar to PerformanceEntry Object
+        // https://developer.mozilla.org/en-US/docs/Web/API/PerformanceEntry
         entries.push({
             name: name,
             duration: Number(duration),
             entryType: 'tailor',
             startTime: perf.now() || Date.now() - perf.timing.navigationStart
         });
     }
+    // Unique API that allows fragments to specify the
+    // time to first meaningul paint of the page
+    function addTTFMPEntry(duration) {
+        addPerfEntry('ttfmp', duration);
```

Comment:
just keeping it as `ttfmp` to avoid the name collision in future once `first-meaningful-paint` is added to (Paint Timing)[https://w3c.github.io/paint-timing/]